File: library/src/main/java/com/example/ruru/library/listener/LoadDataStatus.java
package com.example.ruru.library.listener;

import java.util.List;

/**
 * 加载数据监听器
 */
public interface LoadDataStatus<T> {

    /**
     * 加载数据成功
     */
    void onSuccess(List<T> t);

    /**
     * 加载数据失败
     */
    void onFailure(String msg);

    /**
     * 没有数据了
     */
    void onNoMoreData();
}

File: library/src/main/java/com/example/ruru/library/listener/LoadDataStatus.kt
package com.example.ruru.library.listener

/**
 * 加载数据状态监听器
 */
interface LoadDataStatus<T> {
  /**
   * 加载数据成功
   */
  fun onSuccess(t: List<T>?)

  /**
   * 加载数据失败
   */
  fun onFailure(msg: String?)

  /**
   * 没有数据了
   */
  fun onNoMoreData()
}